util/project_allocation_mapper.py

```diff
@@ -570,8 +571,17 @@ def portal_user_to_tas_obj(self, user, role='Standard'):
                     'department': None,
                     'institution': None,
                     'role': role}
-        return tas_user
 
+        keycloak_client = KeycloakClient()
+        keycloak_user = keycloak_client.get_keycloak_user_by_username(user.username)
+
+        if keycloak_user:
+            for key, val in keycloak_user['attributes'].items():
```

Comment:
In cases like this, where we're mapping between two data types, I'd actually prefer to be explicit:

```python
attrs = keycloak_user['attributes']
tas_user.update({
  'institution': attrs.get('affiliationInstitution'),
  'department': attrs.get('affiliationDepartment'),
  # ... and so on
})
```

The main benefit is that, if I'm coming along later and I want to change the name "affiliationInstitution" to "affiliationUniversity" or something, it's easier for me to find all the places we use that key, because I can simply search for it. Search visibility is a good property when you don't have strict types to fall back on, which normally are what you leverage in such a refactor.